### PR TITLE
Test cases for SCIMGroupHandler class

### PR DESCRIPTION
**Pull request by chirankavinda123**

### CODE REVIEW

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+
+import org.apache.commons.lang.StringUtils;
+import org.junit.rules.ExpectedException;
+import org.mockito.Mock;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.testng.Assert;
+import org.testng.IObjectFactory;
+import org.testng.annotations.*;
```
Comment (sagara-gunathunga) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: Don't use start imports

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+        when(mockedGroupDAO.isExistingGroup(SCIMCommonUtils.getGroupNameWithDomain("GROUP_NAME"),1)).thenReturn(false);
+
+        SCIMGroupHandler scimGroupHandler = new SCIMGroupHandler(1);
+        scimGroupHandler.addMandatoryAttributes(anyString());
+
+    }
+
+    @Test
+    public void testGetGroupAttributesByName() throws Exception {
+        Assert.assertNull(new SCIMGroupHandler(1).getGroupAttributesByName("managers"));
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: better to use static assert imports.

File: components/org.wso2.carbon.identity.scim2.common/pom.xml [hunk 3]
```diff
@@ -204,12 +204,17 @@
                     <suiteXmlFiles>
                         <suiteXmlFile>src/test/resources/testng.xml</suiteXmlFile>
                     </suiteXmlFiles>
+                    <systemPropertyVariables>
+                        <jacoco-agent.destfile>target/jacoco.exec</jacoco-agent.destfile>
+                    </systemPropertyVariables>
                 </configuration>
             </plugin>
+
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.scim2.common/pom.xml]: remove unnecessary new line.

File: components/org.wso2.carbon.identity.scim2.common/pom.xml [hunk 4]
```diff
@@ -204,12 +204,17 @@
...
             <plugin>
                 <groupId>org.jacoco</groupId>
                 <artifactId>jacoco-maven-plugin</artifactId>
                 <version>${jacoco.version}</version>
+
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.scim2.common/pom.xml]: remove unnecessary new line.

File: components/org.wso2.carbon.identity.scim2.common/pom.xml [hunk 5]
```diff
@@ -219,20 +224,22 @@
                     </excludes>
                 </configuration>
                 <executions>
+
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.scim2.common/pom.xml]: remove unnecessary new line.

File: components/org.wso2.carbon.identity.scim2.common/pom.xml [hunk 6]
```diff
@@ -256,13 +263,26 @@
...
+                            <goal>prepare-agent</goal>
+                        </goals>
+                    </execution>
+                    <execution>
+                        <id>default-prepare-agent-integration</id>
+                        <goals>
+                            <goal>prepare-agent-integration</goal>
+                        </goals>
+                    </execution>
+
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.scim2.common/pom.xml]: remove unnecessary new line.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java [hunk 7]
```diff
@@ -0,0 +1,305 @@
...
+import org.wso2.carbon.identity.scim2.common.utils.SCIMCommonUtils;
+import org.wso2.charon3.core.exceptions.CharonException;
+import org.wso2.charon3.core.objects.Group;
+import org.wso2.charon3.core.schema.SCIMConstants;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.text.SimpleDateFormat;
+import java.util.*;
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: avoid using star imports.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java [hunk 8]
```diff
@@ -0,0 +1,305 @@
...
+
+    @Mock
+    private GroupDAO mockedGroupDAO;
+
+    @Mock
+    private Connection connection;
+
+    @Mock
+    private PreparedStatement mockedPreparedStatement;
+
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: remove unnecessary new line

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java [hunk 9]
```diff
@@ -0,0 +1,305 @@
...
+        when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+        when(connection.prepareStatement(anyString())).thenReturn(mockedPreparedStatement);
+        when(mockedPreparedStatement.executeQuery()).thenReturn(resultSet);
+        when(resultSet.next()).thenReturn(false);
+        when(mockedGroupDAO.isExistingGroup(SCIMCommonUtils.getGroupNameWithDomain("NON_EXISTANT_GROUP_NAME"),1)).thenReturn(false);
+
+        SCIMGroupHandler scimGroupHandler = new SCIMGroupHandler(1);
+        scimGroupHandler.createSCIMAttributes(group);
+
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: always assert some conditions/behaviors after the test.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java [hunk 10]
```diff
@@ -0,0 +1,305 @@
...
+        ResultSet resultSet = mock(ResultSet.class);
+        Map<String,String> attributes = new HashMap<String,String>();
+
+        Date date = new Date(2017,10,10,10,10,10);
+
+        attributes.put("urn:ietf:params:scim:schemas:core:2.0:id","100");
+        attributes.put("urn:ietf:params:scim:schemas:core:2.0:meta.created",date.toString());
+        attributes.put("urn:ietf:params:scim:schemas:core:2.0:meta.lastModified",date.toString());
+        attributes.put("urn:ietf:params:scim:schemas:core:2.0:meta.location","colombo");
+
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: remove unnecessary new lines

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java [hunk 11]
```diff
@@ -0,0 +1,305 @@
...
+        when(mockedGroupDAO.isExistingGroup("NON_EXISTING_GROUP_NAME",1)).thenReturn(false);
+        Assert.assertEquals(new SCIMGroupHandler(1).getGroupWithAttributes(group,"NON_EXISTING_GROUP_NAME"),group);
+
+    }
+
+   @Test
+    public void testGetGroupWithAttributesSecondScenario() throws Exception {
+        Group group = new Group();
+        ResultSet resultSet = mock(ResultSet.class);
+
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: remove unnecessary new line.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java [hunk 12]
```diff
@@ -0,0 +1,305 @@
...
+
+       Date today = Calendar.getInstance().getTime();
+       SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
+
+
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: remove unnecessary new line.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java [hunk 13]
```diff
@@ -0,0 +1,305 @@
...
+        when(mockedGroupDAO.isExistingGroup("EXISTING_GROUP_NAME",1)).thenReturn(true);
+
+        whenNew(GroupDAO.class).withNoArguments().thenReturn(mockedGroupDAO);
+        when(mockedGroupDAO.getSCIMGroupAttributes(anyInt(),anyString())).thenReturn(attributes);
+        when(StringUtils.isNotEmpty(resultSet.getString(anyInt()))).thenReturn(true);
+
+        Assert.assertEquals(new SCIMGroupHandler(1).getGroupWithAttributes(group,"EXISTING_GROUP_NAME"),group);
+
+    }
+
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: remove unnecessary new line.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java [hunk 14]
```diff
@@ -0,0 +1,305 @@
...
+        when(resultSet.next()).thenReturn(true);
+        when(mockedPreparedStatement.executeQuery()).thenReturn(resultSet);
+
+        when(mockedGroupDAO.isExistingGroup("EXISTING_GROUP_NAME",1)).thenReturn(true);
+
+        whenNew(GroupDAO.class).withNoArguments().thenReturn(mockedGroupDAO);
+        when(mockedGroupDAO.getSCIMGroupAttributes(anyInt(),anyString())).thenReturn(attributes);
+        when(StringUtils.isNotEmpty(resultSet.getString(anyInt()))).thenReturn(true);
+
+        Assert.assertEquals(new SCIMGroupHandler(1).getGroupWithAttributes(group,"EXISTING_GROUP_NAME"),group);
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: add appropriate messages while asserting

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java [hunk 15]
```diff
@@ -0,0 +1,305 @@
...
+        ResultSet resultSet = mock(ResultSet.class);
+        mockStatic(IdentityDatabaseUtil.class);
+
+        when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+        when(connection.prepareStatement(anyString())).thenReturn(mockedPreparedStatement);
+        when(mockedPreparedStatement.executeQuery()).thenReturn(resultSet);
+        when(resultSet.next()).thenReturn(true);
+
+        SCIMGroupHandler scimGroupHandler = new SCIMGroupHandler(1);
+        scimGroupHandler.deleteGroupAttributes("GROUP_DELETABLE");
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: assert a condition after a test

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java [hunk 16]
```diff
@@ -0,0 +1,305 @@
...
+        mockStatic(IdentityDatabaseUtil.class);
+
+        when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+        when(connection.prepareStatement(anyString())).thenReturn(mockedPreparedStatement);
+        when(mockedPreparedStatement.executeQuery()).thenReturn(resultSet);
+        when(resultSet.next()).thenReturn(true);
+        when(mockedGroupDAO.isExistingGroup("EXISTENT_ROLE_NAME",1)).thenReturn(true);
+
+        SCIMGroupHandler scimGroupHandler = new SCIMGroupHandler(1);
+        scimGroupHandler.updateRoleName("EXISTENT_ROLE_NAME","NEW_ROLE_NAME");
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: assert a condition after a test

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java [hunk 17]
```diff
@@ -0,0 +1,305 @@
...
+        when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+        when(connection.prepareStatement(anyString())).thenReturn(mockedPreparedStatement);
+        when(resultSet.next()).thenReturn(false);
+        when(mockedPreparedStatement.executeQuery()).thenReturn(resultSet);
+        when(mockedGroupDAO.listSCIMGroups()).thenReturn(groups);
+
+        Assert.assertNotNull(new SCIMGroupHandler(1).listSCIMRoles());
+    }
+
+}
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: add a new line

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java [hunk 18]
```diff
@@ -0,0 +1,305 @@
...
+        when(connection.prepareStatement(anyString())).thenReturn(mockedPreparedStatement);
+        when(StringUtils.isNotEmpty(anyString())).thenReturn(true);
+        when(SCIMCommonUtils.getPrimaryFreeGroupName(anyString())).thenReturn("directors");
+        when(mockedPreparedStatement.executeQuery()).thenReturn(resultSet);
+        when(mockedGroupDAO.getGroupNameById(1,"5")).thenReturn("directors");
+
+        Assert.assertEquals(new SCIMGroupHandler(1).getGroupName("5"),"directors");
+
+        when(StringUtils.isNotEmpty(anyString())).thenReturn(false);
+        Assert.assertNull(new SCIMGroupHandler(1).getGroupName("NON_EXISITNG_GROUP_NAME"));
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: better to assert with a message

File: components/org.wso2.carbon.identity.scim2.common/pom.xml (hunk 6 above)
Comment (ashensw) [inline components/org.wso2.carbon.identity.scim2.common/pom.xml]: remove above two executions.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java [hunk 19]
```diff
@@ -0,0 +1,305 @@
...
+
+    }
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+        return new org.powermock.modules.testng.PowerMockObjectFactory();
+    }
+
+    @AfterMethod
+    public void tearDown() throws Exception {
```
Comment (ashensw) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: remove unused methods.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java [hunk 20]
```diff
@@ -0,0 +1,305 @@
...
+        mockStatic(IdentityDatabaseUtil.class);
+        mockStatic(StringUtils.class);
+        when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+        when(connection.prepareStatement(anyString())).thenReturn(mockedPreparedStatement);
+        when(resultSet.next()).thenReturn(false);
+        when(mockedPreparedStatement.executeQuery()).thenReturn(resultSet);
+
+        when(mockedGroupDAO.isExistingGroup("NON_EXISTING_GROUP_NAME",1)).thenReturn(false);
+        Assert.assertEquals(new SCIMGroupHandler(1).getGroupWithAttributes(group,"NON_EXISTING_GROUP_NAME"),group);
+
```
Comment (ashensw) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: remove the unnecessary new line.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java [hunk 21]
```diff
@@ -0,0 +1,305 @@
...
+       when(connection.prepareStatement(anyString())).thenReturn(mockedPreparedStatement);
+       when(resultSet.next()).thenReturn(true);
+       when(mockedPreparedStatement.executeQuery()).thenReturn(resultSet);
+       when(mockedGroupDAO.isExistingGroup("managers",1)).thenReturn(true);
+       Assert.assertTrue(new SCIMGroupHandler(1).isGroupExisting("managers"));
+
+       when(resultSet.next()).thenReturn(false);
+       when(mockedGroupDAO.isExistingGroup("directors",1)).thenReturn(false);
+       Assert.assertEquals(new SCIMGroupHandler(1).isGroupExisting("directors"),false);
+
```
Comment (ashensw) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: remove the unnecessary new line.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java [hunk 22]
```diff
@@ -0,0 +1,305 @@
+package org.wso2.carbon.identity.scim2.common.group;
```
Comment (kasunbg) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: missing license.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java [hunk 23]
```diff
@@ -0,0 +1,305 @@
...
+
+        when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+        when(connection.prepareStatement(anyString())).thenReturn(mockedPreparedStatement);
+        when(mockedPreparedStatement.executeQuery()).thenReturn(resultSet);
+        when(resultSet.next()).thenReturn(true);
+        when(mockedGroupDAO.isExistingGroup(SCIMCommonUtils.getGroupNameWithDomain("ALREADY_EXISTANT_GROUP_NAME"),1)).thenReturn(true);
+
+        SCIMGroupHandler scimGroupHandler = new SCIMGroupHandler(1);
+        scimGroupHandler.createSCIMAttributes(group);
+
```
Comment (kasunbg) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: No asserts in this test!

**Comment by kasunbg** [comment]
Travis has reported a test failure. Please check.

**Comment by kasunbg** [comment]
Travis has reported a test failure. Please check.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java (hunk 1 above)
Comment (chirankavinda123) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: done

File: components/org.wso2.carbon.identity.scim2.common/pom.xml (hunk 3 above)
Comment (chirankavinda123) [reply components/org.wso2.carbon.identity.scim2.common/pom.xml]: done

File: components/org.wso2.carbon.identity.scim2.common/pom.xml (hunk 4 above)
Comment (chirankavinda123) [reply components/org.wso2.carbon.identity.scim2.common/pom.xml]: done

File: components/org.wso2.carbon.identity.scim2.common/pom.xml (hunk 5 above)
Comment (chirankavinda123) [reply components/org.wso2.carbon.identity.scim2.common/pom.xml]: done

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java (hunk 8 above)
Comment (chirankavinda123) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: done

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java (hunk 2 above)
Comment (chirankavinda123) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: done with latest commits

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java (hunk 9 above)
Comment (chirankavinda123) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: done

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java (hunk 10 above)
Comment (chirankavinda123) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: done

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java (hunk 11 above)
Comment (chirankavinda123) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: done

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java (hunk 12 above)
Comment (chirankavinda123) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: done

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java (hunk 13 above)
Comment (chirankavinda123) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: done

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java (hunk 14 above)
Comment (chirankavinda123) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: done

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java (hunk 15 above)
Comment (chirankavinda123) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: done

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java (hunk 16 above)
Comment (chirankavinda123) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: done

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java (hunk 17 above)
Comment (chirankavinda123) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: done

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java (hunk 18 above)
Comment (chirankavinda123) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: done

File: components/org.wso2.carbon.identity.scim2.common/pom.xml (hunk 6 above)
Comment (chirankavinda123) [reply components/org.wso2.carbon.identity.scim2.common/pom.xml]: done

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java (hunk 19 above)
Comment (chirankavinda123) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: removed

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java (hunk 20 above)
Comment (chirankavinda123) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: done

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java (hunk 21 above)
Comment (chirankavinda123) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: done

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java (hunk 22 above)
Comment (chirankavinda123) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java]: added

**Pull request merged**